platform/jewel/markdown/core/src/main/kotlin/org/jetbrains/jewel/markdown/processing/MarkdownProcessor.kt

```diff
@@ -69,14 +74,19 @@ public class MarkdownProcessor(
      * @see DefaultInlineMarkdownRenderer
      */
     public fun processMarkdownDocument(@Language("Markdown") rawMarkdown: String): List<MarkdownBlock> {
-        val blocks =
-            if (editorMode) {
-                processWithQuickEdits(rawMarkdown)
-            } else {
-                parseRawMarkdown(rawMarkdown)
-            }
+        scrollingSynchronizer?.beforeProcessing()
```

Comment:
Nit: the synchronizer could have a `fun doEdit(action: () -> Unit)` that hides the before/after processing inside it